src/Parsers/ExpressionListParsers.cpp

```diff
@@ -1042,4 +1107,1887 @@ bool ParserKeyValuePairsList::parseImpl(Pos & pos, ASTPtr & node, Expected & exp
+            ASTPtr tmp;
+
+            /// Special case for cast expression
+            if (layers.back()->previousType() != OperatorType::TupleElement &&
+                ParseCastExpression(pos, tmp, expected))
+            {
+                layers.back()->pushOperand(std::move(tmp));
+                continue;
+            }
+
+            if (layers.back()->previousType() == OperatorType::Comparison)
+            {
+                SubqueryFunctionType subquery_function_type = SubqueryFunctionType::NONE;
+
+                if (any_parser.ignore(pos, expected) && subquery_parser.parse(pos, tmp, expected))
+                    subquery_function_type = SubqueryFunctionType::ANY;
+                else if (all_parser.ignore(pos, expected) && subquery_parser.parse(pos, tmp, expected))
+                    subquery_function_type = SubqueryFunctionType::ALL;
+
+                if (subquery_function_type != SubqueryFunctionType::NONE)
+                {
+                    Operator prev_op;
+                    ASTPtr function, argument;
+
+                    if (!layers.back()->popOperator(prev_op))
+                        return false;
+                    if (!layers.back()->popOperand(argument))
+                        return false;
+
+                    function = makeASTFunction(prev_op.function_name, argument, tmp);
+
+                    if (!modifyAST(function, subquery_function_type))
+                        return false;
+
+                    layers.back()->pushOperand(std::move(function));
+                    continue;
+                }
+            }
+
+            /// Try to find any unary operators
+            auto cur_op = op_table_unary.begin();
+            for (; cur_op != op_table_unary.end(); ++cur_op)
+            {
+                if (parseOperator(pos, cur_op->first, expected))
+                    break;
+            }
+
+            if (cur_op != op_table_unary.end())
+            {
+                next = Action::OPERAND;
+                layers.back()->pushOperator(cur_op->second);
+                continue;
+            }
+
+            auto old_pos = pos;
+            std::unique_ptr<Layer> layer;
+            if (parseOperator(pos, "INTERVAL", expected))
+                layer = std::make_unique<IntervalLayer>();
+            else if (parseOperator(pos, "CASE", expected))
+                layer = std::make_unique<CaseLayer>();
+
+            /// Here we check that CASE or INTERVAL is not an identifier
+            /// It is needed for backwards compatibility
+            if (layer)
+            {
+                Expected stub;
+
+                auto stub_cur_op = op_table.begin();
+                for (; stub_cur_op != op_table.end(); ++stub_cur_op)
+                {
+                    /// Minus can be unary
+                    /// TODO: check cases 'select case - number from table' and 'select case -x when 10 then 5 else 0 end'
+                    if (stub_cur_op->second.function_name == "minus")
```

Comment:
I don't like comparing with string here.
It is minor, but probably we can remember the position of operator minus in a op_table or something like this.